extmod/asyncio/core.py

```diff
@@ -222,6 +242,9 @@ def run_until_complete(main_task=None):
                 _exc_context["exception"] = exc
                 _exc_context["future"] = t
                 Loop.call_exception_handler(_exc_context)
+                # XXX if we do await it later,
```

Comment:
I'm not sure what this comment means, can you elaborate?

Reply:
`it` in this context is the task. I'll change the comment.

This is a core bug which, before taskgroups, nobody seems to have hit yet.

Reply:
Oh, await'ing a task that's already been await / run. Yes that's known to be not supported I guess, I hadn't thought of it as an issue as such :-)